src/HashBus.Twitter.Monitor.CatchUp/App.cs

```diff
@@ -17,20 +21,46 @@ class App
             string accessToken,
             string accessTokenSecret)
         {
-            var busConfiguration = new BusConfiguration();
-            busConfiguration.Transactions().DoNotWrapHandlersExecutionInATransactionScope();
-            busConfiguration.Transactions().DefaultTimeout(defaultTransactionTimeout);
-            busConfiguration.EndpointName(endpointName);
-            busConfiguration.UseSerialization<JsonSerializer>();
-            busConfiguration.EnableInstallers();
-            busConfiguration.UsePersistence<NHibernatePersistence>().ConnectionString(nserviceBusConnectionString);
-            busConfiguration.ApplyMessageConventions();
-            busConfiguration.RegisterComponents(c=>c.RegisterSingleton<ITweetService>(
+            var endpointConfiguration = new EndpointConfiguration(endpointName);
+            // endpointConfiguration.Transactions().DoNotWrapHandlersExecutionInATransactionScope();
+            endpointConfiguration.UseTransport<MsmqTransport>().Transactions(TransportTransactionMode.ReceiveOnly);
+            endpointConfiguration.UnitOfWork().WrapHandlersInATransactionScope(defaultTransactionTimeout);
+            endpointConfiguration.UseSerialization<JsonSerializer>();
+            endpointConfiguration.EnableInstallers();
+            endpointConfiguration.UsePersistence<NHibernatePersistence>().ConnectionString(nserviceBusConnectionString);
+            endpointConfiguration.ApplyMessageConventions();
+            endpointConfiguration.ApplyErrorAndAuditQueueSettings();
+            endpointConfiguration.RegisterComponents(c=>c.RegisterSingleton<ITweetService>(
                 new TweetService(maximumNumberOfTweetsPerCatchUp, consumerKey, consumerSecret, accessToken, accessTokenSecret)));
 
-            using (Bus.Create(busConfiguration).Start())
+            var transportExtensions = endpointConfiguration.UseTransport<MsmqTransport>();
+            var routing = transportExtensions.Routing();
+            routing.RouteToEndpoint(
+                messageType: typeof(StartCatchUp),
+                destination: "HashBus.Twitter.Monitor.CatchUp");
+
+           routing.RouteToEndpoint(
+                messageType: typeof(AnalyzeTweet),
+                destination: "HashBus.Application");
+
+           routing.RouteToEndpoint(
+                messageType: typeof(TweetReceived),
+                destination: "HashBus.Twitter.Monitor");
+
+
+
+            var endpointInstance = await Endpoint.Start(endpointConfiguration)
+                .ConfigureAwait(false);
+            try
+            {
+                Console.WriteLine("\r\n Bus created and configured; press any key to stop program\r\n");
+
+                Console.ReadKey();

```

Comment:
Same comment as above. Although this endpoint never used `Thread.Sleep`, it should have done. It can now also transition to `await Task.Delay(Timeout.Infinite)`.